src/components/ec-file-upload/ec-file-upload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-file-upload` : 'ec-file-upload'">
+    <label
+      v-if="label || note"
+      class="ec-file-upload__label"
+    >
+      <span
+        v-if="label"
+        data-test="ec-file-upload__label-text"
+        class="ec-file-upload__label-text"
+      >{{ label }}</span>
+
+      <span
+        v-if="note"
+        data-test="ec-file-upload__label-note"
+        class="ec-file-upload__label-note"
+      >{{ note }}</span>
+    </label>
+
+    <ec-file-dropzone
+      data-test="ec-file-upload__dropzone"
+      :is-disabled="isDisabled"
+      @change="onChange"
+    >
+      <slot
+        name="title"
+        slot="title"
+      />
+
+      <slot
+        name="subtitle"
+        slot="subtitle"
+      />
+    </ec-file-dropzone>
+
+    <div
+      data-test="ec-file-upload__help-text"
+      class="ec-file-upload__help-text"
+      :class="{ 'ec-file-upload__help-text--is-disabled': isDisabled }"
+    >
+      <slot name="help-text" />
+    </div>
+
+    <ec-file-list
+      data-test="ec-file-upload__list"
+      :items="fileList"
+      :is-delete-disabled="isDisabled"
+      @delete="onDelete"
+    />
+  </div>
+</template>
+
+<script>
+import EcFileDropzone from '../ec-file-dropzone';
+import EcFileList from '../ec-file-list';
+
+export default {
+  name: 'EcFileUpload',
+  components: {
+    EcFileDropzone,
+    EcFileList,
+  },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    value: {
+      type: Array,
+      default: () => ([]),
+    },
+    label: {
+      type: String,
+    },
+    note: {
+      type: String,
+    },
+    isDisabled: {
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      fileList: [],
```

Comment:
Why is this needed? There is a full list available in `value` prop. You can use the `value` and you can remove the watcher.